pkg/cmd/update-tls-artifacts/generate-owners/tlsmetadatainterfaces/helpers.go

```diff
@@ -23,36 +23,137 @@ func AnnotationValue(whitelistedAnnotations []certgraphapi.AnnotationValue, key
 	errs := []error{}
-	certKeyPairs := certs.SecretInfoByNamespaceName{}
-	caBundles := certs.ConfigMapInfoByNamespaceName{}
+	inClusterCertKeyPairs := certs.SecretInfoByNamespaceName{}
+	onDiskCertsWithoutSecrets := certs.CertKeyPairInfoByOnDiskLocation{}
+	inClusterCABundles := certs.ConfigMapInfoByNamespaceName{}
+	onDiskCABundlesWithoutConfigMaps := certs.CABundleInfoByOnDiskLocation{}
 
 	for i := range rawData {
 		currPKI := rawData[i]
 
 		for i := range currPKI.InClusterResourceData.CertKeyPairs {
 			currCert := currPKI.InClusterResourceData.CertKeyPairs[i]
-			existing, ok := certKeyPairs[currCert.SecretLocation]
+			existing, ok := inClusterCertKeyPairs[currCert.SecretLocation]
 			if ok && !reflect.DeepEqual(existing, currCert.CertKeyInfo) {
 				errs = append(errs, fmt.Errorf("mismatch of certificate info for --namespace=%v secret/%v", currCert.SecretLocation.Namespace, currCert.SecretLocation.Name))
 				continue
 			}
 
-			certKeyPairs[currCert.SecretLocation] = currCert.CertKeyInfo
+			inClusterCertKeyPairs[currCert.SecretLocation] = currCert.CertKeyInfo
 		}
 		for i := range currPKI.InClusterResourceData.CertificateAuthorityBundles {
 			currCert := currPKI.InClusterResourceData.CertificateAuthorityBundles[i]
-			existing, ok := caBundles[currCert.ConfigMapLocation]
+			existing, ok := inClusterCABundles[currCert.ConfigMapLocation]
 			if ok && !reflect.DeepEqual(existing, currCert.CABundleInfo) {
 				errs = append(errs, fmt.Errorf("mismatch of certificate info for --namespace=%v configmap/%v", currCert.ConfigMapLocation.Namespace, currCert.ConfigMapLocation.Name))
 				continue
 			}
 
-			caBundles[currCert.ConfigMapLocation] = currCert.CABundleInfo
+			inClusterCABundles[currCert.ConfigMapLocation] = currCert.CABundleInfo
 		}
+
+		for _, tlsMetadata := range currPKI.OnDiskResourceData.TLSArtifact {
+			found := false
+			isAlsoInCluster := false
+			for _, certKeyPair := range currPKI.CertKeyPairs.Items {
+				for _, loc := range certKeyPair.Spec.OnDiskLocations {
+					if tlsMetadata.OnDiskLocation.Path == loc.Cert.Path || tlsMetadata.OnDiskLocation.Path == loc.Key.Path {
+						found = true
+						isAlsoInCluster = len(certKeyPair.Spec.SecretLocations) > 0
+						break
+					}
+				}
+				if found {
+					break
+				}
+			}
+			if found {
+				if !isAlsoInCluster {
+					info, err := GetCertKeyPairInfoForOnDiskPath(tlsMetadata.OnDiskLocation)
+					if err == nil {
+						onDiskCertsWithoutSecrets[tlsMetadata.OnDiskLocation] = info
+					}
+				}
+				continue
+			}
+
+			for _, caBundle := range currPKI.CertificateAuthorityBundles.Items {
+				for _, loc := range caBundle.Spec.OnDiskLocations {
+					if tlsMetadata.OnDiskLocation.Path == loc.Path {
+						found = true
+						isAlsoInCluster = len(caBundle.Spec.ConfigMapLocations) > 0
+						break
+					}
+				}
+				if found {
+					break
+				}
+			}
+			if found {
+				if !isAlsoInCluster {
```

Comment:
I don't see why it matters whether it's also in the cluster.

Reply:
on-disk-but-also-in-the-cluster artifacts have metadata set in annotations. on-disk-only artifacts need to be looked up in `onDiskCertsWithoutSecrets` / `onDiskCABundlesWithoutConfigMaps` maps. This set needs to be kept minimal as its going to be stored in `origin` (even `api` maybe?)

Reply:
> on-disk-but-also-in-the-cluster artifacts have metadata set in annotations. on-disk-only artifacts need to be looked

I don't think this will work in practice since we see duplicate CA bundles owned by different components.  I suspect we'll end up needing to remove this.